routing/router.go

```diff
@@ -366,8 +366,23 @@ func (a alice) then(h http.Handler) http.Handler {
 func buildHttpHandlerLogger(route Route, handlerFunction http.Handler) http.Handler {
 	c := alice{}
 	c = c.append(hlog.NewHandler(zerolog.With().Str("route", route.Name).Logger()))
-	c = c.append(hlog.HostHandler("Host", true))
-	c = c.append(hlog.RequestIDHandler("RequestID", "Request-Id"))
+	c = c.append(hlog.HostHandler("host", true))
+	c = c.append(hlog.RemoteAddrHandler("remoteAddr"))
+
+	// extract and log the request ID - if we have a X-Request-Id header we use it; otherwise, we generate our own
+	xRequestIdHeader := "X-Request-Id"
+	xRequestFieldKey := "requestId"
+	c = c.append(func(next http.Handler) http.Handler {
+		requestIDGeneratorHandler := hlog.RequestIDHandler(xRequestFieldKey, "")
+		customHeaderHandler := hlog.CustomHeaderHandler(xRequestFieldKey, xRequestIdHeader)
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.Header.Get(xRequestIdHeader) == "" {
+				requestIDGeneratorHandler(next).ServeHTTP(w, r)
+			} else {
+				customHeaderHandler(next).ServeHTTP(w, r)
```

Comment:
Does this still work in other parts of the code if it uses the request specific helper functions like `hlog.IDFromRequest`? Not sure if we use these or not even.

Reply:
We don't use that anywhere. And that IDFromRequest is only used inside hlog to retrieve that generated UUID that is created inside RequestIDHandler - it has nothing to do with the X-Request-ID header (or ANY header for that matter). So don't confuse `IDFromRequest()` as getting anything related to a true request ID. It is only to retrieve the logging infrastructure's generated ID.

See the code here: https://github.com/rs/zerolog/blob/v1.34.0/hlog/hlog.go#L184-L236

Reply:
Right and what we're saying is that there should be no difference between the generated ID and the `X-Request-ID` header since they are both the request id it's just that one is generated internally by Kiali and the other is supplied externally. If we don't get the request ID from the context it is fine then.

Reply:
Yes, for what we are now doing here, we just log some "request ID". The PR is now implemented such that it doesn't matter where that ID came from, its all logged the same (with the name `requestId`).

We don't get the request ID from the context - it will either be generated or it will come from the request header.